/home/brett/anaconda3/lib/python3.8/site-packages/torch/utils/data/dataloader.py:558: UserWarning: This DataLoader will create 10 worker processes in total. Our suggested max number of worker in current system is 4, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(_create_warning_msg(
INFO:root:Epoch 1
INFO:root:Epoch [1/50], Validation Loss: 0.4888
Process Process-1:
Traceback (most recent call last):
  File "/home/brett/anaconda3/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/home/brett/anaconda3/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/brett/Desktop/tutorials/ijepa/ijepa/main_ijepa.py", line 62, in process_main
    app_main(args=params)
  File "/home/brett/Desktop/tutorials/ijepa/ijepa/src/train.py", line 404, in main
    if val_loss < best_val_loss:
UnboundLocalError: local variable 'best_val_loss' referenced before assignment